tests/test_hsm_hotwallet.py

```diff
@@ -0,0 +1,267 @@
+"""Test HSM wallet implementation on Ethereum mainnet fork.
+
+To run tests in this module:
+
+.. code-block:: shell
+
+    export ETH_NODE_URI="https://eth-mainnet.alchemyapi.io/v2/YOUR-API-KEY"
+    export GOOGLE_CLOUD_PROJECT="your-project"
+    export GOOGLE_CLOUD_REGION="us-east1"
+    export KEY_RING="eth-keys"
+    export KEY_NAME="signing-key"
+    export GOOGLE_APPLICATION_CREDENTIALS="/path/to/credentials.json"
+    pytest -k test_hsm_wallet
+"""
+
+import os
+import logging
+import shutil
+
+from cchecksum import to_checksum_address
+import pytest
+from eth_typing import ChecksumAddress, HexAddress
+from web3 import Web3
+
+from eth_defi.provider.anvil import fork_network_anvil
+from eth_defi.provider.multi_provider import create_multi_provider_web3
+from eth_defi.token import fetch_erc20_details
+from eth_defi.tx import decode_signed_transaction
+from eth_defi.hsm_hotwallet import HSMWallet
+
+
+# Skip tests if required env vars are not set
+pytestmark = pytest.mark.skipif(not all([os.environ.get("ETH_NODE_URI"), os.environ.get("GOOGLE_CLOUD_PROJECT"), os.environ.get("GOOGLE_CLOUD_REGION"), os.environ.get("KEY_RING"), os.environ.get("KEY_NAME"), shutil.which("anvil")]), reason="Set ETH_NODE_URI and Google Cloud env vars, and install anvil to run these tests")
+
+# Set up logging
+logger = logging.getLogger(__name__)
+
+
+def setup_module(module):
+    """Set up logging for the test module."""
+    # Get log level from environment, default to WARNING
+    log_level = os.environ.get("LOG_LEVEL", "WARNING").upper()
+    logging.basicConfig(level=log_level, format="%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+
+
+@pytest.fixture()
+def weth_holder() -> ChecksumAddress:
+    """Account that holds a lot of WETH on mainnet."""
+    # Binance 8
+    return to_checksum_address("0xf977814e90da44bfa03b6295a0616a897441acec")
+
+
+@pytest.fixture()
+def anvil_eth_fork(request, weth_holder) -> str:
+    """Create a testable fork of Ethereum mainnet."""
+    mainnet_rpc = os.environ["ETH_NODE_URI"]
+    launch = fork_network_anvil(mainnet_rpc, unlocked_addresses=[weth_holder], fork_block_number=17_500_000)  # Pick a stable recent block
```

Comment:
Instead of fork, you can do a standalone launch, because forks are slower. 

There is nothing on mainnet we need for these tests.

Reply:
You can use standalone Anvil launch, added documentation here:

https://web3-ethereum-defi.readthedocs.io/api/provider/_autosummary_provider/eth_defi.provider.anvil.launch_anvil.html#eth_defi.provider.anvil.launch_anvil